src/pybind/cephfs/cephfs.pyx

```diff
@@ -538,15 +539,28 @@ cdef class LibCephFS(object):
             raise make_ex(ret, "error calling ceph_init")
         self.state = "initialized"
 
-    def mount(self):
+    def mount(self, mount_root=None, filesystem_name=None):
         """
         Perform a mount using the path for the root of the mount.
         """
         if self.state == "configuring":
             self.init()
         self.require_state("initialized")
+
+        default_root = "/"
+        cdef:
+            char *_filesystem_name = filesystem_name
+            char *_mount_root = default_root if mount_root is None else mount_root
```

Comment:
Unfortunately still get this problem:

https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/14029//consoleFull

If I had to guess, it's because the compiler is too simple to be sure the result of the expression is not transient. I think it might be possible to just lift the expression out of the `cdef` which may allow the compiler to proceed. I'll push a commit that tries that.